### PR TITLE
[SPARK-47984][ML][SQL] Change `MetricsAggregate/V2Aggregator#serialize/deserialize` to call `SparkSerDeUtils#serialize/deserialize`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
The utility methods `serialize` and `deserialize` exist in `SparkSerDeUtils`:

https://github.com/apache/spark/blob/08caa567fb29e762f3f7f9d94cd42c02f1e47247/common/utils/src/main/scala/org/apache/spark/util/SparkSerDeUtils.scala#L23-L36

This PR changes the implementation of `serialize/deserialize` methods in `MetricsAggregate` and `V2Aggregator` to call the `serialize/deserialize` methods in `SparkSerDeUtils` to eliminate duplicate code.

### Why are the changes needed?
Eliminate duplicate code.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 1]
```diff
@@ -397,17 +398,12 @@ private[spark] object SummaryBuilderImpl extends Logging {
 
     override def serialize(state: SummarizerBuffer): Array[Byte] = {
       // TODO: Use ByteBuffer to optimize
-      val bos = new ByteArrayOutputStream()
-      val oos = new ObjectOutputStream(bos)
-      oos.writeObject(state)
-      bos.toByteArray
+      Utils.serialize(state)
```
Comment (LuciferYang) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala:401]: https://github.com/apache/spark/blob/08caa567fb29e762f3f7f9d94cd42c02f1e47247/common/utils/src/main/scala/org/apache/spark/util/SparkSerDeUtils.scala#L23-L29

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/V2Aggregator.scala [hunk 2]
```diff
@@ -50,16 +49,11 @@ case class V2Aggregator[BUF <: java.io.Serializable, OUT](
   }
 
   override def serialize(buffer: BUF): Array[Byte] = {
-    val bos = new ByteArrayOutputStream()
-    val out = new ObjectOutputStream(bos)
-    out.writeObject(buffer)
-    out.close()
-    bos.toByteArray
+    Utils.serialize(buffer)
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/V2Aggregator.scala:52]: https://github.com/apache/spark/blob/08caa567fb29e762f3f7f9d94cd42c02f1e47247/common/utils/src/main/scala/org/apache/spark/util/SparkSerDeUtils.scala#L23-L29

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 3]
```diff
@@ -397,17 +398,12 @@ private[spark] object SummaryBuilderImpl extends Logging {
...
     }
 
     override def deserialize(bytes: Array[Byte]): SummarizerBuffer = {
       // TODO: Use ByteBuffer to optimize
-      val bis = new ByteArrayInputStream(bytes)
-      val ois = new ObjectInputStream(bis)
-      ois.readObject().asInstanceOf[SummarizerBuffer]
+      Utils.deserialize(bytes)
```
Comment (LuciferYang) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala:406]: https://github.com/apache/spark/blob/08caa567fb29e762f3f7f9d94cd42c02f1e47247/common/utils/src/main/scala/org/apache/spark/util/SparkSerDeUtils.scala#L32-L36

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala [hunk 4]
```diff
@@ -397,17 +398,12 @@ private[spark] object SummaryBuilderImpl extends Logging {
 
     override def serialize(state: SummarizerBuffer): Array[Byte] = {
       // TODO: Use ByteBuffer to optimize
```
Comment (LuciferYang) [inline mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala:400]: If there is no clear optimization goal, can we remove this TODO now? @zhengruifeng

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

File: mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala (hunk 4 above)
Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala:400]: LGTM

actually this TODO originated from https://github.com/apache/spark/commit/07549b20a3fc2a282e080f76a2be075e4dd5ebc7 , so cc @WeichenXu123

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master. Thank you @LuciferYang and all.

**Comment by LuciferYang** [comment]
Thanks @yaooqinn @dongjoon-hyun @zhengruifeng ~

**Pull request closed without merging**